adc.c

```diff
@@ -0,0 +1,41 @@
+
+#include "system.h"
+#include "adc.h"
+#include "nuts_bolts.h"
+
+#define ADMUX_SELECTION_MASK    0x7
+#define MUX5_MASK               0x8
+
+uint16_t adc_read_channel(uint8_t channel) {
+  // Set ADC reference
+  ADMUX = (1 << REFS0);
+
+  // Enable ADC, Prescaler: 128x
+  ADCSRA = (1 << ADEN) | (1 << ADPS2) | (1 << ADPS1) | (1 << ADPS0);
+
+  // Select Channel
+  ADMUX |= (channel & ADMUX_SELECTION_MASK);
+
+  // Select Channel - ADCSRB, MUX5
+  if (channel & MUX5_MASK) {
+    bit_true(ADCSRB, MUX5_BIT_POS);
+  } else {
+    bit_false(ADCSRB, MUX5_BIT_POS);
+  }
+
+  // Enable ADC capture
+  ADCSRA |= (1 << ADSC);
+
+  // Enable capture of ADC
+  while(ADCSRA & (1 << ADSC));
+
+  // Stote the result
+  uint16_t ret = ADC;
+  
```

Comment:
I'm a little bit concerned about turning the ADC on and off for every sample.  Many ADCs will require some settling time which can lead to weird fluctuations in readings.  Does the datasheet say anything about this?